aspnetcore/signalr/get-started-signalr-core.md

```diff
@@ -0,0 +1,110 @@
+
+1. Add the `Microsoft.AspNetCore.SignalR` NuGet package to the ASP.NET Core project. Select **Tools** > **NuGet Package Manager** > **Package Manager Console**. In the console, issue the following command:
+
+  ```powershell
+  Install-Package Microsoft.AspNetCore.SignalR –IncludePrerelease
+  ```
+
+  The preceeding command installs the [Microsoft.AspNetCore.SignalR](https://www.nuget.org/packages/Microsoft.AspNetCore.SignalR) NuGet package that contains libraries SignalR server-side code.
+
+1. Install the JavaScript client library by using npm. `npm install` creates a *\node_modules\* folder and sub-folders for the libraries in the same location as where you run the npm command.
+
+  ```console
+   npm install @aspnet/signalr-client --save
+  ```
+
+1. Copy the *signalr-client-1.0.0-alpha2-final.js* for SignalR from *\node_modules\@aspnet\signalr-client\dist\browser* to the *wwwroot\lib* folder in your project.
+
+## Create the SignalR Hub
+
+A hub is a class that represents a high-level pipeline that allows the client and server to call methods on each other directly.
+
+1. Add a class to the project by choosing **File** > **New** > **File** and selecting **Visual C# Class**. 
+
+1. Inherit from `Microsoft.AspNetCore.SignalR.Hub`. The `Hub` class contains properties and events for managing connections and groups, as well as sending and receiving data.
+
+1. Create the `Send` method that sends a message to all connected chat clients. Notice it returns a `Task`, because SignalR is asynchronous and scales better.
+
+  [!code-csharp[Startup](get-started-signalr-core/sample/ChatHub.cs?range=7-14)]
+
+## Configure the project to use SignalR
+
+The SignalR server must be configured so that it knows to pass requests to SignalR.
+
+1. To configure a SignalR project, modify the `ConfigureServices` method of the application's `Startup` class by inserting a call to `services.AddSignalR()`.
+
+  This adds SignalR as part of the [ASP.NET Core middleware](xref:fundamentals/middleware/index) pipeline.
+
+1. In the `Startup` class, you must also configure SignalR routes by calling `UseSignalR`. In the call to `UseSignalR` is where you setup the SignalR hub mappings.
+
+  [!code-csharp[Startup](get-started-signalr-core/sample/Startup.cs?highlight=22,40-43)]
+
+## Create the SignalR client code
+
+1. Replace the content in *\Pages\Index.cshtml* with the following code:
+
+  [!code-html[Index](get-started-signalr-core/sample/Index.cshtml)]
+
+  The preceding HTML displays name and message fields, and a submit button. Notice the script references at the bottom: a reference to SignalR and *chat.js*.
+
+1. Add a JavaScript file to the *\wwwroot\js\* folder named *chat.js* and add the following code to it:
+
+  [!code-html[Index](get-started-signalr-core/sample/chat.js)]
+
+## Run the app
+
+1. Select **Debug** > **Start without debugging** to launch a browser and load the website locally.
```

Comment:
Let's instruct the user to open two browser windows (they can use the same browser, just two separate windows) and chat between them. That's the demo that really shows off the ✨magic ✨ of SignalR ;)